scripts/staging-ci/staging-reset.js

```diff
@@ -30,13 +30,15 @@ async function main() {
     printLog(`Staging branch already up to date in ${CI_FILE}. Skipping.`)
   }
 
-  const isStagingAlreadyCreated = await executeCommand(`git ls-remote --heads ${REPOSITORY} ${NEW_STAGING_BRANCH}`)
-  if (!isStagingAlreadyCreated) {
+  const isStagingAlreadyExists = await executeCommand(`git ls-remote --heads ${REPOSITORY} ${NEW_STAGING_BRANCH}`)
+  if (!isStagingAlreadyExists) {
     printLog('Creating the new staging branch...')
     await executeCommand(`git checkout -b ${NEW_STAGING_BRANCH}`)
     await executeCommand(`git push origin ${NEW_STAGING_BRANCH}`)
   } else {
-    printLog('New staging branch already created. Skipping.')
+    printLog('New staging branch already exists, updating...')
+    await executeCommand(`git branch -f ${NEW_STAGING_BRANCH}`)
+    await executeCommand(`git push origin ${NEW_STAGING_BRANCH}`)
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏I don't think this will work as intended:
* `git branch -f staging-nn` will move the branch `staging-nn` reference to the current `HEAD` (which is `main` at this point), but it *won't* switch to the `staging-nn` branch, so `git push` will try to push `main` instead of `staging-nn`. I think what you want is `git checkout -B staging-nn` (uppercase B): https://git-scm.com/docs/git-checkout#Documentation/git-checkout.txt-emgitcheckoutem-b-Bltnew-branchgtltstart-pointgt
* `git push origin` will fail because the upstream `staging-nn` branch is probably not an ancestor of the pushed branch. You might want to use `--force`.

This could work and I'd be fine with it, but why not just removing old branches by hand?